src/block/meson.build

```diff
@@ -0,0 +1,20 @@
+block_swig = custom_target('block',
+    output: [
+        'block_wrap.c',
+        '_block.py',
+    ],
+    input: 'gsl_block.i',
+    command: [swig, '-I../typemaps']  + swig_flags + [
+            '-o', '@OUTPUT0@', '-outdir', '@OUTDIR@', '@INPUT@'],
+    install: true,
+    install_dir: py.get_install_dir() / 'pygsl',
```

Comment:
You probably don't want to install the C source. Check out the details for https://mesonbuild.com/Reference-manual_functions.html#custom_target_install_dir regarding how to install only a single output for a multi-output custom target.